File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        dic = {}
        res = 0
        for num in nums:
            if num not in dic:
                dic[num] = dic.get(num - 1, 0) + dic.get(num + 1, 0) + 1
                dic[num - dic.get(num - 1, 0)] = dic[num]
                dic[num + dic.get(num + 1, 0)] = dic[num]
                res = max(res, dic[num])
        return res

File: Data Structures & Algorithms/maximum-subarray/submission-4.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int res{1 << 31};
        int sum{};
        for (const auto& num : nums) {
            if (sum < 0) sum = 0;
            sum += num;
            res = max(res, sum);
        }
        return res;
    }
};
